components/director/internal/domain/application/service.go

```diff
@@ -764,6 +765,15 @@ func (s *service) Merge(ctx context.Context, destID, srcID string) (*model.Appli
 		return nil, errors.Errorf("Application templates are not the same. Destination app template: %s. Source app template: %s", destTemplateID, srcTemplateID)
 	}
 
+	_, ok := srcAppLabels[xsappnameKey]
+	if ok {
+		return nil, errors.Errorf("Source app template: %s has label xsappname", srcTemplateID)
+	}
+	_, ok = destAppLabels[xsappnameKey]
```

Comment:
Probably we can add an e2e test that registers 2 systems with that label and tries to merge them. We can assert that the merge mutation has failed. Don't forget to use the label key from an env variable again.